### DISCUSSION TITLE
How to fix "A drive with the name '.C' does not exist." error

### DISCUSSION THREAD

**Discussion by zirubak** [PromtEngineer/localGPT#498] [General]
Do you guys know how to fix this error? 

Cannot find drive. A drive with the name '.C' does not exist.
At C:\Users\baek\OneDrive - HP Inc\Documents\WindowsPowerShell\profile.ps1:4 char:5
+ If (Test-Path "C:\Users\baek\AppData\Local\anaconda3\Scripts\conda.ex ...
+     ~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (.C:String) [], DriveNotFoundException
    + FullyQualifiedErrorId : DriveNotFound

**Comment by tvan97** [comment]
same problem...
I'm using cmder temporarily
did you fix this?

**Comment by klein-davis** [comment]
I am having the same issue. Vs Code was working fine for me, and then started throwing this error about 2 hours into my coding session. I did not install any packages, and I didn't even use the terminal. I was just using the debug code button to run my code, but then something changed and now it has stopped working. Pip is also having trouble installing packages now as well. I think it has something to do with the C:\Users\\{username}\Documents\WindowsPowerShell\profile.ps1 profile that is managed by conda. But I wasn't changing anything in conda at the time, so I am completely clueless. I'm not even using localGPT (though I am doing stuff with local llms), but googling only lead me here, so here I am.

**Comment by Sheriff1980** [comment]
Same issue here, right after I installed the Github autocode thing

**Comment by klein-davis** [comment]
### Temporary fix:
As VS Code is first loading the terminal of your project, the error will show up almost immediately. But after a few moments, it will disappear for like three seconds. During the time where it is not present, type "pip" and press enter. For me, this prevents the error and returns my terminal to working order until I open a new terminal or reopen VS Code.

As far as permanent fixes go, I suspected it was GitLens, since I had been fiddling with it around the time that the error showed up. But removing every non-required plugin still resulted in the error: ```Cannot find drive. A drive with the name '.C' does not exist.``` citing whichever command I try to run.

**Comment by vsoraas** [comment]
If you guys have cloned the repos into folders that have a hyphen in them, try removing hyphen from folder name.

**Reply by klein-davis** [reply]
I did have my project in a folder where the filepath had hyphens. But I renamed it so that there are no hyphens, and it still gives the same error:
```Test-Path : Cannot find drive. A drive with the name '.C' does not exist.```
The current filepath for my project: ```D:\Local Language Assistant```
I was skeptical that it was the filepath since I have had the same project location for the past few weeks.

**Comment by jmdfgouveia1** [comment]
Executing any command using PowerShell on VSCode now returns this error for me. Interestingly, when I use PowerShell on desktop or any other terminal on VSCode, it works fine.

Any suggestions as to what the issue is?

**Reply by kmljt** [reply]
any fix?

**Comment by tvan97** [comment]
try to choose default terminal -> enter the folder path of powershell in you pc and check if it works or not
i changed sth, including these steps above and now every time i run the code, it displays 'python' terminal instead of powershell
anyway it's ok now, without error

**Comment by vsoraas** [comment]
Problem is likely with the Python extension and how it plays with environments (maybe conda specifically) - it was for me. Fix described here:
https://github.com/microsoft/vscode-python/issues/22073#issuecomment-1734290798

TL;DR: switch the python extension to pre-release version and see if that helps.

**Reply by klein-davis** [reply]
That seems to have worked for me. Thank you!

**Reply by bobbyng626** [reply]
That worked for me too, thanks.

**Reply by TonyH0401** [reply]
Thank you, this solution works nicely.
